mmeval/core/dist_backends/oneflow_dist.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+import numpy as np
+import pickle
+from typing import TYPE_CHECKING, Any, List, Tuple, TypeVar, Union
+
+from mmeval.utils import try_import
+from .base_backend import TensorBaseDistBackend
+
+if TYPE_CHECKING:
+    import oneflow
+    import oneflow as flow
+    import oneflow.framework.check_point_v2 as check_point_v2
+else:
+    flow = try_import('oneflow')
+    check_point_v2 = try_import('oneflow.framework.check_point_v2')
+
+Tensor = TypeVar('Tensor', bound='oneflow.Tensor')
+
+
+class OneFlowDist(TensorBaseDistBackend):
+    """A distributed communication backend for oneflow."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        if flow is None:
+            raise ImportError(f'For availability of {self.__class__.__name__},'
+                              ' please install oneflow first.')
+
+    @property
+    def is_initialized(self) -> bool:
+        """Returns True if the distributed environment has been initialized.
+
+        Returns:
+            bool: Returns True if the distributed environment has been
+            initialized, otherwise returns False.
+        """
+        try:
+            flow.env.get_world_size()
+            is_init = True
+        except ValueError:
+            is_init = False
+        return is_init
+
+    @property
+    def rank(self) -> int:
+        """Returns the rank index of the current process group."""
+        return flow.env.get_local_rank()
+
+    @property
+    def world_size(self) -> int:
+        """Returns the world size of the current process group."""
+        return flow.env.get_world_size()
+
+    def _object_to_tensor(self, obj: Any) -> Tuple[Tensor, Tensor]:
+        """Convert the given object to a tensor via `pickle.dumps`.
+
+        Args:
+            obj (any): Any pickle-able python object.
+
+        Returns:
+            Tuple: A tuple of the tensor converted from given object and the
+            tensor size.
+        """
+        device = 'cuda'
+        if flow.cuda.device_count() == 0:
+            device = 'cpu'
+        buffer = pickle.dumps(obj)
+        storage = np.frombuffer(buffer, dtype=np.int8)
+        tensor = flow.tensor(storage).to(device)
+        local_size_tensor = flow.tensor([tensor.numel()], device=device)
+        return tensor, local_size_tensor
```

Comment:
Why do we want to convert a CUDA Tensor here? And what are the problems with using a CPU Tensor? 🧐